src/main/resources/public/js/components/dropdown_menus/instance/add-instance-dropdown-header.js

```diff
@@ -53,19 +64,68 @@ class AddInstanceDropdownHeader extends React.Component {
             });
     }
 
+    filterApps = (app) => {
+        const org = this.props.organization;
+        // only applications
+        if (AppTypes.application !== app.type) {
+            return false;
+        }
+
+        //Check types
+        return (app.target_publicbodies && org.type === 'PUBLIC_BODY') ||
+            (app.target_companies && org.type === 'COMPANY') ||
+            (app.target_citizens && !org.type);
+    }
+
+    _fetchApplications = async () => {
+        const {apps, maybeMoreApps, totalAppsFetched} = this.state
+        const {apps: appsFetched, maybeMoreApps: maybeMoreAppsFetched}  = await customFetch(`/api/store/applications`, {urlParams: {
```

Comment:
at this stage, I think you know the org type so you could already set the correct `target_*` parameters ?

Reply:
maybe it would also be easier to filter on the `app.type` in the backend